src/app/pages/vendor-dashboard/components/vendor-filter/vendor-filter.component.ts

```diff
@@ -1,15 +1,95 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import {COMMA, ENTER, N} from '@angular/cdk/keycodes';
+import { MatChipInputEvent } from '@angular/material/chips';
+import { FormBuilder, FormGroup } from '@angular/forms';
+
+export interface workType{
+  name: string;
+  
+}
+
+
 
 @Component({
   selector: 'app-vendor-filter',
   templateUrl: './vendor-filter.component.html',
   styleUrls: ['./vendor-filter.component.scss']
 })
-export class VendorFilterComponent implements OnInit {
+export class VendorFilterComponent{
+
+  addOnBlur = true;
+  readonly separatorKeysCodes = [ENTER, COMMA] as const;
+  workTypes: workType[] = [];  
+
+  statuses = new Map(
+    [['inContract', false], 
+    ['Active', false], 
+    ['Inactive', false], 
+    ['hasIssues', false]]
+    );
+
+  
+  constructor() {}
+
+
+  @Output() newItemEvent = new EventEmitter<string>();
+
+  addNewItem(value: string) {
+    this.newItemEvent.emit(value);
+    this.workTypes.push({name: value});
+    
+  }
+
+  add(event: MatChipInputEvent): void {
+    const value = (event.value || '').trim();
+
+    // Add our fruit
+    if (value) {
+      this.workTypes.push({name: value});
+    }
+
+    // Clear the input value
+    event.chipInput!.clear();
+  }
+
+  remove(type: workType): void {
+    const index = this.workTypes.indexOf(type);
+
+    if (index >= 0) {
+      this.workTypes.splice(index, 1);
+    }
+  }
+  checBoxChange(completed: boolean, name: string): void{
+      this.statuses.set(name, completed);
+      // console.log(name, this.statuses.get(name))
+  }
+
+  filterApply(): any{
+
+    const output = [
+      this.workTypes,
+      this.statuses
+    ]
 
-  constructor() { }
 
-  ngOnInit(): void {
+    console.log(output)
+    
+    return this.workTypes
```

Comment:
this is where the eventEmitter should be used. You would emit(query) where the query object is what you are currently calling output. And it should be a query object not an array.

Reply:
The object can be defined in the types vendor.ts file because we know what the query should look like.
```
export interface VendorQuery {
 text: string,
 status:  StatusTypes[] // this will need to be defined as the enum like I was talking about above in contract, active, etc.
work_type: WorkType // should be a defined entity in the types/work-type.ts file 
sort: "ASC" | "DESC" | -1 | 1
}
```